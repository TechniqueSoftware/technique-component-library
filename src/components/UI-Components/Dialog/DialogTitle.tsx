import * as React from 'react';
import { withStyles, createStyles, WithStyles, Theme } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      margin: 0,
      padding: theme.spacing(2),
    },
    closeButton: {
      position: 'absolute',
      right: theme.spacing(1),
      top: theme.spacing(1),
      color: theme.palette.grey[500],
    },
  });
export interface DialogTitleProps extends WithStyles<typeof styles> {
  id: string;
  children: React.ReactNode;
  onClose: () => void;
}

export const DialogTitle = withStyles(styles)((props: DialogTitleProps) => {
  const { id, children, classes, onClose, ...other } = props;
  return (
      <MuiDialogTitle disableTypography className={classes.root} {...other}>
        <Typography variant="h4" component="h2">{children}</Typography>
        {onClose ? (
          <IconButton id={`${id}-close-button`} aria-label="close" className={classes.closeButton} onClick={onClose}>
            <CloseIcon />
          </IconButton>
        ) : null}
      </MuiDialogTitle>
  );
});
