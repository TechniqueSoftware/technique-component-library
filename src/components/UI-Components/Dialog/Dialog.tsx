import * as React from 'react';
import MuiDialog, { DialogProps as MuiDialogProps } from '@material-ui/core/Dialog';
import { DialogTitle } from './DialogTitle';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Divider from '@material-ui/core/Divider';
import { ButtonWithIconProps } from '../Buttons/ButtonWithIcon/ButtonWithIcon';
import ButtonWithIcon from '../Buttons/ButtonWithIcon/ButtonWithIcon';

export interface DialogProps {
  id: string;
  dialogTitle:string;
  open:boolean;
  children?: React.ReactNode;
  fullScreen?: boolean;
  fullWidth?: boolean;
  maxWidth?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | false;
  primaryButtonProps?: ButtonWithIconProps;
  secondaryButtonProps: ButtonWithIconProps;
  onClose : () => void;
}
export default function Dialog(props: DialogProps) {
  const {
    id,
    fullWidth,
    maxWidth,
    fullScreen,
    dialogTitle,
    open,
    onClose,
    primaryButtonProps,
    secondaryButtonProps,
    children
  } = props;

  const muiDialogProps: MuiDialogProps = {
    open,
    fullWidth,
    maxWidth,
    fullScreen,
  };

  const primaryButtonPropsWithDefaults: ButtonWithIconProps = {
    color: 'primary',
    ...primaryButtonProps,
  };

  const secondaryButtonPropsWithDefaults: ButtonWithIconProps = {
    color: 'primary',
    variant: 'text',
    ...secondaryButtonProps
  };

  return (
    <div>
      <MuiDialog
        id={`${id}-dialog`}
        onClose={onClose}
        aria-labelledby={`${id}-dialog-title`}
        {...muiDialogProps}
      >
        <DialogTitle id={`${id}-dialog-title`} onClose={onClose}>
          {dialogTitle}
        </DialogTitle>
        <Divider orientation="horizontal" />
        <DialogContent id={`${id}-dialog-content`}>
          {children}
        </DialogContent>
        <Divider orientation="horizontal" />
        <DialogActions id={`${id}-dialog-actions`}>
          {secondaryButtonProps && <ButtonWithIcon {...secondaryButtonPropsWithDefaults} />}
          {primaryButtonProps && <ButtonWithIcon {...primaryButtonPropsWithDefaults} />}
        </DialogActions>
      </MuiDialog>
    </div>);
}
