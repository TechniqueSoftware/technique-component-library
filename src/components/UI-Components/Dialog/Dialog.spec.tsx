import Dialog, { DialogProps } from './Dialog';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('Dialog', () => {

  test('should render with base props', () => {
    const baseProps: DialogProps = {
      children: <span>mockChildren</span>,
      primaryButtonProps: { onClick: jest.fn() },
      secondaryButtonProps: { onClick: jest.fn() },
      dialogTitle: 'Mock Dialog Title',
      id:'mockId',
      open: false,
      fullScreen: false,
      fullWidth: false,
      maxWidth: 'xs',
      onClose: jest.fn()
    };
    const shallowWrapper = shallow(<Dialog {...baseProps} />);
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });
});
