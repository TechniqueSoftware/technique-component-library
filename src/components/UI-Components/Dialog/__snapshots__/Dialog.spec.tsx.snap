// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render with base props 1`] = `
"<div>
  <WithStyles(ForwardRef(Dialog)) id=\\"mockId-dialog\\" onClose={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} aria-labelledby=\\"mockId-dialog-title\\" open={false} fullWidth={false} maxWidth=\\"xs\\" fullScreen={false}>
    <WithStyles(Component) id=\\"mockId-dialog-title\\" onClose={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }}>
      Mock Dialog Title
    </WithStyles(Component)>
    <WithStyles(ForwardRef(Divider)) orientation=\\"horizontal\\" />
    <WithStyles(ForwardRef(DialogContent)) id=\\"mockId-dialog-content\\">
      <span>
        mockChildren
      </span>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(Divider)) orientation=\\"horizontal\\" />
    <WithStyles(ForwardRef(DialogActions)) id=\\"mockId-dialog-actions\\">
      <ButtonWithIcon color=\\"primary\\" variant=\\"text\\" onClick={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} />
      <ButtonWithIcon color=\\"primary\\" onClick={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} />
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>"
`;
